### PR TITLE
Local plugin support/documentation

### LINKED ISSUE #1507: Support for Local Plugin Files in Addition to HTTPS
**Is your feature request related to a problem? Please describe.**
I'm trying to write a custom Megalinter plugin that's defined in a private repository, meaning a GET request to the .yml definition cannot be made without setting access tokens in request headers.

**Describe the solution you'd like**
Support for file paths in the PLUGINS property within `.mega-linter.yml`. For example:

```yml
PLUGINS:
  - "<private_repo_submodule>/mega-linter-plugin-xxx/xxx.megalinter-descriptor.yml"
```

**Describe alternatives you've considered**
Support for the `file://` protocol. For example:

```yml
PLUGINS:
  - file://<pwd>/mega-linter-plugin-xxx/xxx.megalinter-descriptor.yml
```

**Additional context**
I would be willing to work on this. Thanks!

### PR DESCRIPTION
**Pull request by jackheuberger**
Fixes #1507 

## Proposed Changes

1. Support for local plugin descriptor files under a custom "file://" header under the PLUGINS section of `.mega-linter.yml`
2. Documentation in the plugin section for the new schema

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Review by nvuillam — CHANGES_REQUESTED** [review]
Great ! :)
Please could you add:
- a test class checking use of file plugin (if possible)
- Update README.md, because plugins.md if automatically generated from it :)
Thank a lot !

**Comment by jackheuberger** [comment]
I think I changed everything correctly. Let me know if there's anything else!

**Review by jackheuberger — COMMENTED** [review]
Made the change in README, added a test in plugins_test.py

**Comment by nvuillam** [comment]
Thanks 
Please can you also merge main into your branch then merge conflicts ? ( those 3 files are automatically generated so they way you merge them is not important ^^ )

**Comment by jackheuberger** [comment]
Ok I think I fixed everything. Lmk if I messed something up somewhere but I think I did it correctly

File: megalinter/tests/test_megalinter/plugins_test.py [hunk 1]
```diff
@@ -23,12 +23,13 @@ def test_load_plugin_success(self):
...
             local_repo = Repo(search_parent_directories=True)
             local_branch = local_repo.active_branch.name
+
         except:  # noqa: E722
             local_branch = "master"
         mega_linter, output = utilstest.call_mega_linter(
             {
                 "PLUGINS": f"https://raw.githubusercontent.com/{ML_REPO}/"
-                + local_branch
+                + "main"
```
Comment (jackheuberger) [inline megalinter/tests/test_megalinter/plugins_test.py:32]: I hard-coded main in this instance as {ML_REPO} wasn't successfully pointing to my fork

File: .automation/test/mega-linter-plugin-test/test.megalinter-descriptor.yml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-    examples:
-      - "jsonlint myfile.jsonc"
-    install:
-      dockerfile:
-        - RUN npm install jsonlint -g && echo "This is my test MegaLinter plugin !"
+- examples:
+  - jsonlint myfile.jsonc
+  install:
+    dockerfile:
+    - RUN npm install jsonlint -g && echo "This is my test MegaLinter plugin !"
```
Comment (Kurt-von-Laven) [inline .automation/test/mega-linter-plugin-test/test.megalinter-descriptor.yml:11]: This is a minor pre-existing issue, but there shouldn't be a space before the exclamation point.

**Review by Kurt-von-Laven — CHANGES_REQUESTED** [review]

File: megalinter/tests/test_megalinter/plugins_test.py (hunk 1 above)
Comment (jackheuberger) [reply megalinter/tests/test_megalinter/plugins_test.py:32]: Sounds good. I wrote a little blurb as to why the change was made with a note to maybe change in the future.

**Review by nvuillam — APPROVED** [review]
Thanks for your contribution :)

**Pull request merged**